crawl-ref/source/tilemcache.cc

```diff
@@ -1243,7 +1243,10 @@ bool mcache_monster::valid(const monster_info& mon)
                              && Options.tile_shield_offsets.first != INT_MAX)
         || get_shield_offset(mon_tile, &ox, &oy);
     return (mon.inv[MSLOT_WEAPON] && have_weapon_offs)
-        || (mon.inv[MSLOT_SHIELD] && have_shield_offs);
+            || ((mon.inv[MSLOT_SHIELD]
+                    || (mon.inv[MSLOT_ALT_WEAPON]
+                            && mons_class_wields_two_weapons(mon.type)))
+                 && have_shield_offs);
```

Comment:
If the position of `have_shield_offs` is right in this condition (not sure just from reading) then it probably merits a code comment, because that implies that `get_shield_offset` is used for second weapons as well as shields. (Is it? I couldn't tell by just reading that function either, so if so, it could use an edit to the docstring as well.)

Reply:
It is, yes.